tests/unit/s2n_connection_blinding_test.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#define ONE_S INT64_C(1000000000)
+
+S2N_RESULT s2n_connection_calculate_blinding(struct s2n_connection *conn, int64_t *min, int64_t *max);
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    struct {
+        uint32_t custom_blinding;
+        uint64_t expected_min;
+        uint64_t expected_max;
+    } test_cases[] = {
+        { .custom_blinding = 0, .expected_min = 0, .expected_max = 0 },
+        { .custom_blinding = 1, .expected_min = 333333333, .expected_max = 1000000000 },
+        { .custom_blinding = 3, .expected_min = 1000000000, .expected_max = 3000000000 },
+        { .custom_blinding = 30, .expected_min = S2N_DEFAULT_BLINDING_MIN * ONE_S, .expected_max = S2N_DEFAULT_BLINDING_MAX * ONE_S },
+        { .custom_blinding = UINT32_MAX, .expected_min = (UINT32_MAX * ONE_S) / 3, .expected_max = UINT32_MAX * ONE_S },
```

Comment:
My concern around UINT32_MAX would be overflow in the math, because I have a hard time conceptualizing of the scale of UINT64_MAX :) I'm pretty sure UINT32_MAX s < UINT64_MAX ns, but please prove it to me. Repeating the same math doesn't prove correctness.

Reply:
Yes this is safe. UINT32_MAX * 10^9 = 4.2 * 10^18. Whereas UINT64_MAX = 1.8 * 10^19. Or [this link might be easier to parse](https://www.wolframalpha.com/input?i=2%5E32+*+%2810%5E9%29+%3C+2%5E64)